src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1442,20 +1442,12 @@ void PhaseIdealLoop::insert_pre_post_loops(IdealLoopTree *loop, Node_List &old_n
     }
   }
 
-  // Nodes inside the loop may be control dependent on a predicate
-  // that was moved before the preloop. If the back branch of the main
-  // or post loops becomes dead, those nodes won't be dependent on the
-  // test that guards that loop nest anymore which could lead to an
-  // incorrect array access because it executes independently of the
-  // test that was guarding the loop nest. We add a special CastII on
-  // the if branch that enters the loop, between the input induction
-  // variable value and the induction variable Phi to preserve correct
-  // dependencies.
```

Comment:
Noticed that this comment block should have been removed earlier with [JDK-8334724](https://bugs.openjdk.org/browse/JDK-8334724) which removed the cast node:

I squeezed this in here - probably not worth a separate task.